### ISSUE TITLE
[SPIKE-EQUIPO-B] Implementar hedonic pricing model (Gràcia)

### ISSUE THREAD

**Issue by prototyp33** [prototyp33/barcelona-housing-demographics-analyzer#208]
## Parent Epic
#198

## Objetivo
Implementar modelo de precios hedónico para el distrito de Gràcia y validar su capacidad predictiva (R² ≥0.55).

## Descripción
Desarrollar modelo de regresión lineal múltiple (OLS) que explique precios de alquiler/venta en Gràcia usando variables demográficas y económicas. Este modelo será la base del analytics engine de v2.0.

## Contexto del Spike
Esta issue es parte del **Spike de Validación (Dec 16-20)**:
- **Equipo:** Equipo B (Analytics & Modeling)
- **Owner:** Tech Lead Data Science
- **Timeline:** Lunes Dec 16 - Viernes Dec 20

### Día 1-2: Setup & EDA (Lunes-Martes)
- [ ] Setup Jupyter notebook environment
- [ ] Recibir sample data de Equipo A (martes EOD)
- [ ] Exploratory Data Analysis (EDA)
  - [ ] Distribuciones de variables
  - [ ] Correlaciones
  - [ ] Outliers detection
- [ ] Feature engineering inicial
  - [ ] Normalización/estandarización
  - [ ] Variables derivadas (ratios, logs)

### Día 3: Modeling (Miércoles)
- [ ] Baseline OLS model
  - [ ] Selección de variables independientes
  - [ ] Train/test split (80/20)
  - [ ] Fit baseline model
- [ ] Diagnostics
  - [ ] Multicolinealidad (VIF)
  - [ ] Heteroskedasticidad (Breusch-Pagan)
  - [ ] Normalidad de residuales (Shapiro-Wilk)
- [ ] Iteración 1: Ajustar variables

### Día 4: Refinamiento (Jueves)
- [ ] Iteración 2-3: Feature selection
  - [ ] Backward/forward selection
  - [ ] Regularización (Ridge/Lasso) si necesario
- [ ] Cross-validation (k=5)
- [ ] Análisis de coeficientes
  - [ ] Signos esperados (precio ↑ con renta ↑)
  - [ ] Significancia estadística (p<0.05)
  - [ ] Magnitudes razonables

### Día 5: Validación & Reporting (Viernes)
- [ ] Test final en test set
- [ ] Generar visualizaciones
  - [ ] Actual vs Predicted scatter plot
  - [ ] Residuals plot
  - [ ] Coefficients bar chart
- [ ] Escribir model performance report
- [ ] Preparar presentación para GO decision meeting (3 PM)

## Dependencies

**Depends On:**
- #199 (Portal Dades) - SOFT BLOCKER (necesita datos demográficos martes EOD)
- #200 (INE API) - SOFT BLOCKER (necesita datos económicos martes EOD)
- #206 (Validation Framework) - SOFT (puede usar framework o implementar propio)

**Workaround si datos reales se retrasan:**
- Lunes-martes AM: Usar mock data/datos históricos para prototyping
- Martes PM: Swap a datos reales cuando Equipo A entregue

**Blocks:**
- #198 (Spike Epic) - GO decision depende del R² del modelo

**Related To:**
- #203 (DiD viability) - Enfoque alternativo/complementario

**Feeds Into:**
- #204 (PostgreSQL schema) - Granularidad del modelo informa diseño de fact tables

### GO Criteria (Must meet for GO decision)
✅ **R² ≥ 0.55** en test set (explained variance)  
✅ **≥5 variables significativas** (p<0.05)  
✅ **Coeficientes con signos esperados**:
   - precio_alquiler ↑ con renta_media ↑
   - precio_alquiler ↓ con desempleo ↑
   - precio_alquiler ↑ con educacion_universitaria ↑  
✅ **Residuales razonablemente normales** (QQ-plot aceptable)  
✅ **No heteroskedasticidad severa** (Breusch-Pagan p>0.05 o corregible)  
✅ **Resultados replicables** (notebook ejecutable end-to-end)

### NO-GO Triggers (Any of these → NO-GO)
❌ R² < 0.40 (modelo explica muy poco)  
❌ Solo 1-2 variables significativas  
❌ Coeficientes contradictorios (precio ↑ con desempleo ↑)  
❌ Multicolinealidad severa (VIF>10 en variables clave)  
❌ Resultados no estables (high variance entre train/test)

### CAVEAT (GO with adjustments)
⚠️ 0.40 ≤ R² < 0.55: GO pero con plan de mejora en v2.0  
⚠️ 3-4 variables significativas: GO pero buscar variables adicionales  
⚠️ Heteroskedasticidad: GO pero usar robust standard errors

### 1. Jupyter Notebook
**File:** `spike-data-validation/notebooks/02-hedonic-model-gracia.ipynb`

**Estructura:**
```python
# 1. Setup & Data Loading
# 2. Exploratory Data Analysis
# 3. Feature Engineering
# 4. Model Training (OLS)
# 5. Diagnostics
# 6. Validation
# 7. Results & Interpretation
# 8. Conclusions
```

### 2. Model Performance Report
**File:** `spike-data-validation/outputs/reports/model_performance_report.md`

**Contenido:**
- Executive summary (R², key findings)
- Model specifications (variables, equation)
- Coefficients table (values, p-values, interpretation)
- Diagnostics results (VIF, heteroskedasticity, normality)
- Validation metrics (train R², test R², RMSE, MAE)
- GO/NO-GO recommendation

### 3. Visualizations
**Files:** `spike-data-validation/outputs/visualizations/`
- `model_coefficients.png` - Bar chart de coeficientes con CI
- `actual_vs_predicted.png` - Scatter plot
- `residuals_plot.png` - Residuals vs fitted
- `qq_plot.png` - Q-Q plot de residuales

### 4. Serialized Model (Optional)
**File:** `spike-data-validation/models/hedonic_model_gracia_v1.pkl`
- Modelo guardado para reuso futuro

## Tech Stack

```python
# Required packages
pandas>=2.0.0
numpy>=1.24.0
statsmodels>=0.14.0  # OLS regression
scikit-learn>=1.3.0  # train_test_split, metrics
matplotlib>=3.7.0
seaborn>=0.12.0
scipy>=1.11.0  # statistical tests
```

### Variable Dependiente
- `precio_alquiler` (€/mes) o `precio_venta` (€/m²)

### Variables Independientes (Preliminary)

**Demográficas (Portal Dades #199):**
- `poblacion_total`
- `edad_media`
- `pct_extranjeros`
- `densidad_poblacional`

**Económicas (INE API #200):**
- `renta_media_hogar` (€/año) - **KEY VARIABLE**
- `tasa_desempleo` (%)
- `pct_estudios_superiores` (%)

**Geográficas (Catastro #201):**
- `distancia_centro` (km)
- `area_barrio` (km²)

**Derivadas (Feature Engineering):**
- `log_renta_media`
- `densidad_norm` (z-score)
- `ratio_empleo` (1 - tasa_desempleo)

### Martes EOD (Dec 17, 5:00 PM)
**Receive from Equipo A:**
- CSV con datos limpios de Gràcia (10-20 barrios × variables)
- Data dictionary (nombre columna → significado)
- Notas de calidad de datos

**Deliver to Equipo A:**
- Lista de variables adicionales necesarias (si aplica)
- Feedback sobre granularidad de datos

### Jueves PM (Dec 19, 3:00 PM) - Joint Sync
**Present:**
- Modelo baseline (R², coeficientes)
- Diagnostics findings
- Preliminary GO/NO-GO signal

**Discuss:**
- ¿Modelo necesita datos adicionales de Equipo A?
- ¿Schema v2.0 (#204) soporta granularidad del modelo?

### Viernes PM (Dec 20, 3:00 PM) - GO Decision Meeting
**Present:**
- Final model performance (R², test metrics)
- Coefficients interpretation
- GO/NO-GO recommendation (model perspective)
- Next steps for v2.0 if GO

## Estimated Hours
**12 hours** over 5 days (2-3h/day average)

**Breakdown:**
- Setup & EDA: 3h
- Modeling: 4h
- Validation: 2h
- Reporting & viz: 2h
- Presentation prep: 1h

## References

- **Architecture Doc:** `docs/spike/reference/ARQUITECTURA_DATOS_VARIABLES.md`
- **33 Variables Mapping:** `docs/spike/reference/mapeo_33_variables_completo.csv`
- **Validation Framework:** Issue #206
- **Academic References:** 
  - Rosen (1974) - Hedonic Prices and Implicit Markets
  - Sheppard (1999) - Hedonic analysis of housing markets

---

## Custom Fields (configurar en UI después de crear)
- **Status:** Backlog
- **Priority:** P0
- **Size:** L
- **Estimate:** 12
- **Epic:** AN
- **Release:** Spike
- **Phase:** Planning
- **Start Date:** 2025-12-16
- **Target Date:** 2025-12-20
- **Quarter:** Q4 2025
- **Effort (weeks):** 0.3
- **Outcome:** Hedonic model with R² ≥0.55 for Gràcia district

---

## Team Assignment
- **Team:** Equipo B (Analytics & Modeling)
- **Owner:** @[TECH_LEAD_DATA_SCIENCE]
- **Reviewer:** Tech Lead (code review before GO meeting)
